ibis/backends/base/sql/alchemy/datatypes.py

```diff
@@ -33,6 +33,22 @@ def get_col_spec(self, **_):
         return f"STRUCT({pairs})"
 
 
+class UInt64(sa.types.Integer):
```

Comment:
This opts for creating a single unsigned integer type used across all backends (with specialization handled by the `compiles` decorator), rather than one specific to the duckdb backend. I don't know if this is a good idea, but it seems to work.

Note that mysql also supports unsigned integers, and the mysql dialect has its own unsigned integer sqlalchemy types that we may want to use for mysql instead of our own custom ones here. See e.g. https://docs.sqlalchemy.org/en/14/dialects/mysql.html#sqlalchemy.dialects.mysql.BIGINT.__init__.

Reply:
It would mean going from the general case here to a more duckdb-specialized one, but I wonder if we should upstream this bit into duckdb_engine?

Reply:
It would be useful in duckdb_engine regardless of what we decided to do here I think, but I'm not sure we should necessarily take that on in any form other than to make [an upstream issue in duckdb_engine](https://github.com/Mause/duckdb_engine/issues/329)